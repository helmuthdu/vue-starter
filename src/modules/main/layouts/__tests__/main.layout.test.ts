import { mount, VueWrapper } from '@vue/test-utils';
import { createMemoryHistory, createRouter } from 'vue-router';
import MainLayout from '../main.layout.vue';

const router = createRouter({ history: createMemoryHistory(), routes: [] });

describe('Layout -> Main', () => {
  let wrapper: VueWrapper<any>;

  beforeEach(() => {
    wrapper = mount(MainLayout, {
      global: {
        plugins: [router],
        stubs: ['router-view'],
      },
    });
  });

  it('should renders with props', () => {
    expect(wrapper.exists()).toBe(true);
  });
});
