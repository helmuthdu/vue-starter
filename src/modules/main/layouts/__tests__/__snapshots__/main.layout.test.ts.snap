// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout -> Home should renders with props 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": null,
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div>
        <section
          class="hero is-medium is-primary"
        >
          <div
            class="hero-body"
          >
            <div
              class="container"
            >
              <div
                class="columns"
              >
                <div
                  class="column is-8-desktop is-offset-2-desktop"
                >
                  <h1
                    class="title is-2 is-spaced"
                  >
                    Hello Bulma!
                  </h1>
                  <h2
                    class="subtitle is-4"
                  >
                     Congratulations! You're running the 
                    <strong>
                      Bulma start
                    </strong>
                     project. 
                    <br />
                     It includes everything you need to 
                    <strong>
                      build your own website
                    </strong>
                     with Bulma. 
                  </h2>
                </div>
              </div>
            </div>
          </div>
        </section>
        <section
          class="section"
        >
          <div
            class="container"
          >
            <!---->
          </div>
        </section>
        <footer
          class="footer has-text-centered"
        >
          <div
            class="container"
          >
            <div
              class="columns"
            >
              <div
                class="column is-8-desktop is-offset-2-desktop"
              >
                <p>
                  <strong
                    class="has-text-weight-semibold"
                  >
                    <a
                      href="https://www.npmjs.com/package/bulma-start"
                    >
                      bulma-start@0.0.3
                    </a>
                  </strong>
                </p>
                <p>
                  <small>
                     Source code licensed 
                    <a
                      href="http://opensource.org/licenses/mit-license.php"
                    >
                      MIT
                    </a>
                  </small>
                </p>
                <p
                  style="margin-top:1rem;"
                >
                  <a
                    href="http://bulma.io"
                  >
                    <img
                      alt="Made with Bulma"
                      height="24"
                      src="made-with-bulma.png"
                      width="128"
                    />
                  </a>
                </p>
              </div>
            </div>
          </div>
        </footer>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "RouterLink": Object {
          "name": "RouterLink",
          "props": Object {
            "activeClass": [Function],
            "ariaCurrentValue": Object {
              "default": "page",
              "type": [Function],
            },
            "custom": [Function],
            "exactActiveClass": [Function],
            "replace": [Function],
            "to": Object {
              "required": true,
              "type": Array [
                [Function],
                [Function],
              ],
            },
          },
          "setup": [Function],
        },
        "RouterView": Object {
          "__emits": null,
          "__props": Array [
            Object {
              "name": Object {
                "0": false,
                "1": true,
                "default": "default",
                "type": [Function],
              },
              "route": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
            },
            Array [
              "name",
            ],
          ],
          "inheritAttrs": false,
          "name": "RouterView",
          "props": Object {
            "name": Object {
              "0": false,
              "1": true,
              "default": "default",
              "type": [Function],
            },
            "route": [Function],
          },
          "setup": [Function],
        },
        "transition": Object {
          "name": "transition",
          "props": undefined,
          "render": [Function],
        },
        "transition-group": Object {
          "name": "transition-group",
          "props": undefined,
          "render": [Function],
        },
      },
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {
          "$router": Object {
            "addRoute": [Function],
            "afterEach": [Function],
            "back": [Function],
            "beforeEach": [Function],
            "beforeResolve": [Function],
            "currentRoute": RefImpl {
              "__v_isRef": true,
              "_rawValue": Object {
                "fullPath": "/",
                "hash": "",
                "matched": Array [],
                "meta": Object {},
                "name": undefined,
                "params": Object {},
                "path": "/",
                "query": Object {},
                "redirectedFrom": undefined,
              },
              "_shallow": true,
              "_value": Object {
                "fullPath": "/",
                "hash": "",
                "matched": Array [],
                "meta": Object {},
                "name": undefined,
                "params": Object {},
                "path": "/",
                "query": Object {},
                "redirectedFrom": undefined,
              },
            },
            "forward": [Function],
            "getRoutes": [Function],
            "go": [Function],
            "hasRoute": [Function],
            "install": [Function],
            "isReady": [Function],
            "onError": [Function],
            "options": Object {
              "history": Object {
                "base": "",
                "createHref": [Function],
                "destroy": [Function],
                "go": [Function],
                "listen": [Function],
                "location": "",
                "push": [Function],
                "replace": [Function],
                "state": Object {},
              },
              "routes": Array [],
            },
            "push": [Function],
            "removeRoute": [Function],
            "replace": [Function],
            "resolve": [Function],
          },
        },
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {
        Symbol([vue-router]: router): Object {
          "addRoute": [Function],
          "afterEach": [Function],
          "back": [Function],
          "beforeEach": [Function],
          "beforeResolve": [Function],
          "currentRoute": RefImpl {
            "__v_isRef": true,
            "_rawValue": Object {
              "fullPath": "/",
              "hash": "",
              "matched": Array [],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
            "_shallow": true,
            "_value": Object {
              "fullPath": "/",
              "hash": "",
              "matched": Array [],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
          },
          "forward": [Function],
          "getRoutes": [Function],
          "go": [Function],
          "hasRoute": [Function],
          "install": [Function],
          "isReady": [Function],
          "onError": [Function],
          "options": Object {
            "history": Object {
              "base": "",
              "createHref": [Function],
              "destroy": [Function],
              "go": [Function],
              "listen": [Function],
              "location": "",
              "push": [Function],
              "replace": [Function],
              "state": Object {},
            },
            "routes": Array [],
          },
          "push": [Function],
          "removeRoute": [Function],
          "replace": [Function],
          "resolve": [Function],
        },
        Symbol([vue-router]: route location): Object {
          "fullPath": "/",
          "hash": "",
          "matched": Array [],
          "meta": Object {},
          "name": undefined,
          "params": Object {},
          "path": "/",
          "query": Object {},
          "redirectedFrom": undefined,
        },
        Symbol([vue-router]: router view location): RefImpl {
          "__v_isRef": true,
          "_rawValue": Object {
            "fullPath": "/",
            "hash": "",
            "matched": Array [],
            "meta": Object {},
            "name": undefined,
            "params": Object {},
            "path": "/",
            "query": Object {},
            "redirectedFrom": undefined,
          },
          "_shallow": true,
          "_value": Object {
            "fullPath": "/",
            "hash": "",
            "matched": Array [],
            "meta": Object {},
            "name": undefined,
            "params": Object {},
            "path": "/",
            "query": Object {},
            "redirectedFrom": undefined,
          },
        },
      },
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {
        "$router": Object {
          "addRoute": [Function],
          "afterEach": [Function],
          "back": [Function],
          "beforeEach": [Function],
          "beforeResolve": [Function],
          "currentRoute": RefImpl {
            "__v_isRef": true,
            "_rawValue": Object {
              "fullPath": "/",
              "hash": "",
              "matched": Array [],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
            "_shallow": true,
            "_value": Object {
              "fullPath": "/",
              "hash": "",
              "matched": Array [],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
          },
          "forward": [Function],
          "getRoutes": [Function],
          "go": [Function],
          "hasRoute": [Function],
          "install": [Function],
          "isReady": [Function],
          "onError": [Function],
          "options": Object {
            "history": Object {
              "base": "",
              "createHref": [Function],
              "destroy": [Function],
              "go": [Function],
              "listen": [Function],
              "location": "",
              "push": [Function],
              "replace": [Function],
              "state": Object {},
            },
            "routes": Array [],
          },
          "push": [Function],
          "removeRoute": [Function],
          "replace": [Function],
          "resolve": [Function],
        },
      },
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.11",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "hasOwnProperty": [Function],
  },
  "rootVM": Object {},
  "wrapperElement": <div>
    <section
      class="hero is-medium is-primary"
    >
      <div
        class="hero-body"
      >
        <div
          class="container"
        >
          <div
            class="columns"
          >
            <div
              class="column is-8-desktop is-offset-2-desktop"
            >
              <h1
                class="title is-2 is-spaced"
              >
                Hello Bulma!
              </h1>
              <h2
                class="subtitle is-4"
              >
                 Congratulations! You're running the 
                <strong>
                  Bulma start
                </strong>
                 project. 
                <br />
                 It includes everything you need to 
                <strong>
                  build your own website
                </strong>
                 with Bulma. 
              </h2>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="section"
    >
      <div
        class="container"
      >
        <!---->
      </div>
    </section>
    <footer
      class="footer has-text-centered"
    >
      <div
        class="container"
      >
        <div
          class="columns"
        >
          <div
            class="column is-8-desktop is-offset-2-desktop"
          >
            <p>
              <strong
                class="has-text-weight-semibold"
              >
                <a
                  href="https://www.npmjs.com/package/bulma-start"
                >
                  bulma-start@0.0.3
                </a>
              </strong>
            </p>
            <p>
              <small>
                 Source code licensed 
                <a
                  href="http://opensource.org/licenses/mit-license.php"
                >
                  MIT
                </a>
              </small>
            </p>
            <p
              style="margin-top:1rem;"
            >
              <a
                href="http://bulma.io"
              >
                <img
                  alt="Made with Bulma"
                  height="24"
                  src="made-with-bulma.png"
                  width="128"
                />
              </a>
            </p>
          </div>
        </div>
      </div>
    </footer>
  </div>,
}
`;
